### PR TITLE
[ABW-3529/3267] Linked Connectors updates

### PR DESCRIPTION
**Pull request by matiasbzurovski**
Jira ticket: [ABW-3529](https://radixdlt.atlassian.net/browse/ABW-3529)
Jira ticket: [ABW-3267](https://radixdlt.atlassian.net/browse/ABW-3267)

## Description
Updates the UI on `Linked Connectors` view, as well as adding the ability to rename them.

## Video

https://github.com/user-attachments/assets/264066f7-ba05-4d2c-a3c4-ec15deb8bfc7

[ABW-3529]: https://radixdlt.atlassian.net/browse/ABW-3529?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ABW-3267]: https://radixdlt.atlassian.net/browse/ABW-3267?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: RadixWallet/Features/P2PLinksFeature/UpdateP2PLinkName/UpdateP2PLinkName+View.swift [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		if sanitizedName != nil {
+			(.enabled, nil)
+		} else {
+			(.disabled, .iconError(L10n.LinkedConnectors.RenameConnector.errorEmpty))
+		}
+	}
+}
+
+extension UpdateP2PLinkName {
+	public struct ViewState: Equatable {
```
Comment (danvleju-rdx) [inline RadixWallet/Features/P2PLinksFeature/UpdateP2PLinkName/UpdateP2PLinkName+View.swift:25]: Do we need `ViewState` here? Couldn't we use `UpdateP2PLinkName.State` directly?

File: RadixWallet/Features/P2PLinksFeature/UpdateP2PLinkName/UpdateP2PLinkName+View.swift [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+				.withNavigationBar {
+					dismiss()
+				}
+				.presentationDetents([.fraction(0.55)])
+				.presentationDragIndicator(.visible)
+				.presentationBackground(.blur)
+		}
+
+		private var content: some SwiftUI.View {
+			WithViewStore(store, observe: \.viewState, send: { .view($0) }) { viewStore in
```
Comment (danvleju-rdx) [inline RadixWallet/Features/P2PLinksFeature/UpdateP2PLinkName/UpdateP2PLinkName+View.swift:54]: Have you considered using `WithPerceptionTracking`?

**Review by danvleju-rdx — APPROVED** [review]
LGTM, left some questions

File: RadixWallet/Features/P2PLinksFeature/Coordinator/P2PLinksFeature+Reducer.swift [hunk 3]
```diff
@@ -91,15 +93,21 @@ public struct P2PLinksFeature: Sendable, FeatureReducer {
...
 		}
 	}
 
 	public func reduce(into state: inout State, internalAction: InternalAction) -> Effect<Action> {
 		switch internalAction {
 		case let .loadLinksResult(.success(linksFromProfile)):
-			state.links = .init(
-				uniqueElements: linksFromProfile.map { P2PLinkRow.State(link: $0) }
-			)
+			state.links = .init(uniqueElements: linksFromProfile)
```
Comment (GhenadieVP) [inline RadixWallet/Features/P2PLinksFeature/Coordinator/P2PLinksFeature+Reducer.swift:110]: Should be able o use `linksFromProfile.asIdentifiable()`, preferred to `.init(uniqueElements:` which can potentially crash.

File: RadixWallet/Clients/RadixConnectClient/RadixConnectClient+Interface.swift
Comment (GhenadieVP) [inline RadixWallet/Clients/RadixConnectClient/RadixConnectClient+Interface.swift:1]: Updates to RadixConnectClient should not be needed, use P2PLinksClient directly.

File: RadixWallet/Features/P2PLinksFeature/UpdateP2PLinkName/UpdateP2PLinkName+View.swift [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+		init(store: StoreOf<UpdateP2PLinkName>) {
+			self.store = store
+		}
+
+		public var body: some SwiftUI.View {
+			content
+				.withNavigationBar {
+					dismiss()
+				}
+				.presentationDetents([.fraction(0.55)])
```
Comment (GhenadieVP) [inline RadixWallet/Features/P2PLinksFeature/UpdateP2PLinkName/UpdateP2PLinkName+View.swift:20]: Should have also the `.large` added.

File: RadixWallet/Features/P2PLinksFeature/UpdateP2PLinkName/UpdateP2PLinkName+View.swift [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+		@FocusState private var textFieldFocus: Bool
+
+		init(store: StoreOf<UpdateP2PLinkName>) {
+			self.store = store
+		}
+
+		public var body: some SwiftUI.View {
+			content
+				.withNavigationBar {
+					dismiss()
```
Comment (GhenadieVP) [inline RadixWallet/Features/P2PLinksFeature/UpdateP2PLinkName/UpdateP2PLinkName+View.swift:18]: We do usually send the dismiss to the reducer and call dismiss from there.

**Review by GhenadieVP — APPROVED** [review]

**Pull request merged**